src/hotspot/share/runtime/os.cpp

```diff
@@ -1675,7 +1675,13 @@ char* os::attempt_reserve_memory_at(char* addr, size_t bytes, bool executable) {
   return result;
 }
 
+static void assert_nonempty_range(const char* addr, size_t bytes) {
+  assert(addr != nullptr && bytes > 0, "invalid range (" PTR_FORMAT ", " SIZE_FORMAT ")",
```

Comment:
The message suggests it is *range*, i.e. `(begin, end)`. I wonder if we better print `p2i(addr) + bytes`?

Reply:
Okay, I changed the message

Reply:
I think `p2i(addr + bytes)` is UB when `addr` is `nullptr`? `p2i(addr) + bytes` is safer?